roofit/hs3/test/histfactory.py

```diff
@@ -0,0 +1,178 @@
+        ws = self.toWS(meas)
+        ws_from_js = self.importToWS("hf.js", "ws1")
+
+        noset = ROOT.RooArgSet()
+
+        mc = ws.obj("ModelConfig")
+        assert mc
+
+        mc_from_js = ws_from_js.obj("meas_modelConfig")
+        assert mc_from_js
+
+        pdf = mc.GetPdf()
+        assert pdf
+
+        pdf_from_js = ws_from_js.pdf(meas.GetName())
+        assert pdf_from_js
+
+        data = ws.data("obsData")
+        assert data
+
+        data_from_js = ws_from_js.data("obsData")
+        assert data_from_js
+
+        result = pdf.fitTo(
+            data,
+            ROOT.RooFit.Save(),
+            ROOT.RooFit.Strategy(1),
+            ROOT.RooFit.Minos(mc.GetParametersOfInterest()),
+            ROOT.RooFit.GlobalObservables(mc.GetGlobalObservables()),
+        )
+
+        #        ROOT.RooMsgService.instance().addStream(ROOT.RooFit.DEBUG)
+        result_from_js = pdf_from_js.fitTo(
+            data_from_js,
+            ROOT.RooFit.Save(),
+            ROOT.RooFit.Strategy(1),
+            ROOT.RooFit.Minos(mc_from_js.GetParametersOfInterest()),
+            ROOT.RooFit.GlobalObservables(mc_from_js.GetGlobalObservables()),
+        )
+
+        from math import isclose
+
+        assert isclose(
+            ws.var("mu").getVal(),
+            ws_from_js.var("mu").getVal(),
+            rel_tol=1e-4,
+            abs_tol=1e-4,
+        )
+        assert isclose(
+            ws.var("mu").getError(),
+            ws_from_js.var("mu").getError(),
+            rel_tol=1e-4,
+            abs_tol=1e-4,
+        )
+
+    def test_closure_loop(self):
+        meas = self.measurement()
+        ws = self.toWS(meas)
+
+        tool = ROOT.RooJSONFactoryWSTool(ws)        
+        js = tool.exportJSONtoString()
+        tool.exportJSON("tmp.js")
+        
+        newws = ROOT.RooWorkspace("new");
+        newtool = ROOT.RooJSONFactoryWSTool(newws)
+        newtool.importJSONfromString(js)
+        
+        mc = ws.obj("ModelConfig")
+        assert mc
+
+        newws.Print()
+        newmc = newws.obj("simPdf_modelConfig")
+        assert newmc
+
+        pdf = mc.GetPdf()
+        assert pdf
+
+        newpdf = newws.pdf("simPdf")
+        assert newpdf
+
+        data = ws.data("obsData")
+        assert data
+
+        newdata = newws.data("obsData")
+        assert newdata
+        
+        result = pdf.fitTo(
+            data,
+            ROOT.RooFit.Save(),
+            ROOT.RooFit.Strategy(1),
+            ROOT.RooFit.Minos(mc.GetParametersOfInterest()),
+            ROOT.RooFit.GlobalObservables(mc.GetGlobalObservables()),
+        )
+
+        #        ROOT.RooMsgService.instance().addStream(ROOT.RooFit.DEBUG)
+        newresult = newpdf.fitTo(
+            newdata,
+            ROOT.RooFit.Save(),
+            ROOT.RooFit.Strategy(1),
+            ROOT.RooFit.Minos(newmc.GetParametersOfInterest()),
+            ROOT.RooFit.GlobalObservables(newmc.GetGlobalObservables()),
+        )
```

Comment:
```suggestion
        newpdf.fitTo(
            newdata,
            Strategy=1,
            Minos=newmc.GetParametersOfInterest(),
            GlobalObservables=newmc.GetGlobalObservables(),
            PrintLevel=-1,

        )
```
* no need to use `Save()` if you don't use the result
* make test less verbose by using `PrintLevel=-`
* you can use the new keyword argument Pythonizations to make the test less verbose